src/components/toolbars/toolbar/toolbar/toolbar.tsx

```diff
@@ -283,17 +285,14 @@ export class Toolbar implements ComponentInterface {
     const range: Range | undefined = selection?.getRangeAt(0);
     const rect: DOMRect | undefined = range?.getBoundingClientRect();
 
-    const containerRect: DOMRect | undefined = this.containerRef?.getBoundingClientRect();
+    //const containerRect: DOMRect | undefined = this.containerRef?.getBoundingClientRect();
+    const styloContainerRect = (this.tools.parentNode as ShadowRoot).host.getBoundingClientRect();
 
     const eventX: number = unifyEvent(this.anchorEvent).clientX;
     const eventY: number = unifyEvent(this.anchorEvent).clientY;
 
-    const x: number =
-      rect && containerRect
-        ? rect.left - containerRect.left + this.containerRef.offsetLeft + rect.width / 2
-        : eventX;
-    const y: number =
-      rect && containerRect ? rect.top - containerRect.top + this.containerRef.offsetTop : eventY;
+    const x: number = eventX - styloContainerRect.x;
```

Comment:
does not work if edges are reached

<img width="1307" alt="Capture d’écran 2022-02-14 à 17 13 05" src="https://user-images.githubusercontent.com/16886711/153903107-39b1f770-e702-4152-85fd-47e3cc4e7f6b.png">
<img width="1106" alt="Capture d’écran 2022-02-14 à 17 18 31" src="https://user-images.githubusercontent.com/16886711/153903120-37f46aa7-0672-42e4-a65b-87b3b13957a7.png">

Reply:
okay, can you help with this? 
I use stylo inside of other webcomponents, so I need to position relative to them.
And also the stylon container is in another component then the edited content

Reply:
you can have a look further in the function, that's where overflow and safearea are handled: https://github.com/papyrs/stylo/blob/f9b3d0085a5d0f8bf7f894a040cab0c0567cac86/src/components/toolbars/toolbar/toolbar/toolbar.tsx#L314

Reply:
I see more problems here...
using window or screen width.
who says my container where the contenteditable is in is as big as the page?
for example we use this: https://github.com/node-projects/dock-spawn-ts
so every component is much smaler as the screen

Reply:
almost but not there yet 

- it needs an offset for the positioning y because the toolbar covers often the selection
- when a all paragraph is selected the toolbar's position feels random. it should be top centered

<img width="1307" alt="Capture d’écran 2022-02-16 à 17 33 16" src="https://user-images.githubusercontent.com/16886711/154312799-f610ffda-377d-4c2d-bbad-153633b99d67.png">

<img width="1052" alt="Capture d’écran 2022-02-16 à 17 43 56" src="https://user-images.githubusercontent.com/16886711/154314130-a9e79697-54f3-481f-b8aa-0e4cb312a0f9.png">
<img width="1052" alt="Capture d’écran 2022-02-16 à 17 44 00" src="https://user-images.githubusercontent.com/16886711/154314135-df9d4f10-b6d4-4111-95b7-00198419cc70.png">